r4/smartconfig/smart_configuration_generator.bal

```diff
@@ -14,37 +14,68 @@
+public isolated function generateSmartConfiguration() returns models:SmartConfiguration|error {
+    utils:LogDebug("Generating smart configuration started");
 
-    public isolated function init() {
-        self.issueHandler = new ("SmartConfigurationGenerator");
-        self.logHandler = new ("SmartConfigurationGenerator");
-        self.logHandler.Debug("Generating smart configuration initialized");
+    models:OpenIDConfiguration openIdConfigurations = {};
+    string? discoveryEndpoint = configs.discoveryEndpoint;
+    if discoveryEndpoint is string && discoveryEndpoint != "" {
+        openIdConfigurations = check utils:getOpenidConfigurations(discoveryEndpoint).cloneReadOnly();
+    } else {
+        utils:LogDebug(string `${constants:VALUE_NOT_FOUND}: discoveryEndpoint`);
     }
 
-    # Generator function for Smart Configuration
-    # + return - smart configuration as a json or an error
-    public isolated function generate() returns models:SmartConfiguration|error {
-        self.logHandler.Debug("Generating smart configuration started");
+    string? authorizationEndpoint = configs.smartConfiguration?.authorizationEndpoint?:openIdConfigurations.authorization_endpoint?:();
+    if authorizationEndpoint is () || authorizationEndpoint == "" {
+        return error(string `${constants:VALUE_NOT_FOUND}: Authorization endpoint`);
+    }
 
-        models:SmartConfiguration|error smartConfig = {
-            ...smart_configuration
-        };
+    string? tokenEndpoint = configs.smartConfiguration?.tokenEndpoint?:openIdConfigurations.token_endpoint?:();
+    if tokenEndpoint is () || tokenEndpoint == "" {
+        return error(string `${constants:VALUE_NOT_FOUND}: Token endpoint`);
+    }
 
-        if smartConfig is error {
-            self.issueHandler.addServiceError(createServiceError(constants:ERROR, constants:VALUE, smartConfig));
-        }
+    string[]? capabilities = configs.smartConfiguration?.capabilities?:();
+    if capabilities is () || capabilities.length() == 0 {
+        return error(string `${constants:VALUE_NOT_FOUND}: Capabilities`);
+    }
 
-        self.logHandler.Debug("Completed generating smart configuration");
-        return smartConfig;
+    string[]? codeChallengeMethodsSupported = configs.smartConfiguration?.codeChallengeMethodsSupported?:openIdConfigurations.code_challenge_methods_supported?:();
+    if codeChallengeMethodsSupported is () || codeChallengeMethodsSupported.length() == 0 {
+        return error(string `${constants:VALUE_NOT_FOUND}: Code challenge methods supported`);
     }
+
+    string[]? grantTypesSupported = configs.smartConfiguration?.grantTypesSupported?:openIdConfigurations.grant_types_supported?:();
+    if grantTypesSupported is () || grantTypesSupported.length() == 0 {
+        return error(string `${constants:VALUE_NOT_FOUND}: Grant types supported`);
+    }
+
+    models:SmartConfiguration smartConfig = {
+        authorization_endpoint: authorizationEndpoint, 
```

Comment:
Rename the variable name to match with the field name and just use `authorization_endpoint` which is equivalent to `authorization_endpoint: authorization_endpoint`. Update the other fields as well.

Reply:
In that case lets rename the field in `SmartConfiguration` to `authorizationEndpoint`

Reply:
The issue is this `SmartConfiguration` is a standardized payload -> https://build.fhir.org/ig/HL7/smart-app-launch/conformance.html#sample-response. Using the same naming patterns in the record as in the standard definition, we can directly serialize and deserialize payload.

Reply:
That makes sense. Let's ignore the best practice and rename the variable to `authorization_endpoint` since it is a requirement

Reply:
Noted. Addressed changes in https://github.com/ballerina-platform/module-ballerinax-health.fhir.templates/commit/cc5624bf0a90f2825cbee979b74aa06a84fe401f